### PR TITLE
IK moves hips

### PR DESCRIPTION
**Pull request by AndrewMeadows**
The inverse kinematics (IK) system computes a hip offset as a function of head and other IK targets, then shifts the skeleton's hips accordingly.  Works in 2D-screenie mode and also with HMD.

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Build console output for commit: 6eb6608b9d66bcfe21ae1bff83595e0db6e11ddd available at: https://gist.github.com/c64ba0c32d71a63e1d5f<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6031.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6031.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 7bbb57ae183a61d9f6fe522ebff57d2333558c6a available at: https://gist.github.com/c64ba0c32d71a63e1d5f<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6031.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6031.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: e1a5afcff317bf58190c8a52b34af624aebaa449 available at: https://gist.github.com/c64ba0c32d71a63e1d5f<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6031.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6031.dmg

File: interface/resources/meshes/defaultAvatar_full/avatar-animation.json [hunk 1]
```diff
@@ -51,7 +65,7 @@
                         "id": "spineLean",
                         "type": "manipulator",
                         "data": {
-                            "alpha": 1.0,
+                            "alpha": 0.0,
                             "joints": [
```
Comment (AndrewMeadows) [inline interface/resources/meshes/defaultAvatar_full/avatar-animation.json]: I had to disable the **spineLean** manipulator.  Maybe we should completely remove it at this stage?

File: interface/resources/meshes/defaultAvatar_full/avatar-animation.json [hunk 2]
```diff
@@ -16,24 +16,38 @@
...
-                            "typeVar": "headAndNeckType"
+                            "typeVar": "neckType"
                         },
                         {
                             "jointName": "Head",
                             "positionVar": "headPosition",
                             "rotationVar": "headRotation",
-                            "typeVar": "headAndNeckType"
+                            "typeVar": "headType"
                         }
```
Comment (AndrewMeadows) [inline interface/resources/meshes/defaultAvatar_full/avatar-animation.json]: The IK target type of head and neck are no longer necessarily the same... depending on using HMD or not.

**Comment by hifi-gustavo** [comment]
Build console output for commit: 22786dea7d3d3438888fe46cbadc15215b5b5fce available at: https://gist.github.com/c64ba0c32d71a63e1d5f<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6031.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6031.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 8ebdd25b7123e9620545471ac9783ca9a66e4c79 available at: https://gist.github.com/c64ba0c32d71a63e1d5f<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6031.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6031.dmg

File: interface/resources/meshes/defaultAvatar_full/avatar-animation.json (hunk 1 above)
Comment (hyperlogic) [inline interface/resources/meshes/defaultAvatar_full/avatar-animation.json]: We can for now, but I would like to preserve the twist part of the spine lean.  This should make the shoulders twist with the head.  Which is more natural.  But yes, I agree the foward/back left/right lean should be removed.

Comment (AndrewMeadows) [inline interface/resources/meshes/defaultAvatar_full/avatar-animation.json]: I believe the shoulders do now twist when turning head in HMD mode... at least it should work when HMD and hydras are being used at the same time, but not necessarily when the hydras are in their cradle -- the underpose targets for the hands will probably lock the shoulders in place.  We'll probably have to do something else for "hanging arms at idle" than to target the underpose.

File: libraries/animation/src/Rig.cpp [hunk 3]
```diff
@@ -1070,9 +1073,11 @@ void Rig::updateNeckJoint(int index, const HeadParameters& params) {
 
                 _animVars.unset("headPosition");
                 _animVars.set("headRotation", realLocalHeadOrientation);
-                _animVars.set("headAndNeckType", QString("RotationOnly"));
+                _animVars.set("headAndNeckType", (int)IKTarget::Type::RotationOnly);
```
Comment (hyperlogic) [inline libraries/animation/src/Rig.cpp]: I think we can just delete the setting of the "headAndNeckType" animVar as it is no longer in the current avatar-animation.json.

**Pull request merged**